ddtrace/ddtrace.go

```diff
@@ -89,8 +101,8 @@ type FinishConfig struct {
 	// finishing.
 	Error error
 
-	// NoDebugStack will prevent any set errors from generating an attached stack trace tag.
-	NoDebugStack bool
+	// ErrorConfig is configuration for stack trace attached to error
+	ErrorConfig
```

Comment:
I like your solution and I like the idea of including the fields here, in `FinishConfig`. But I'm not sure the `ErrorConfig` makes sense as part of the API, given that it's only here to serve a private method (`setTagError`). You can keep the fields if you like, but please don't use an embedded structure for this and rather write them here directly.

Maybe the private `errorConfig` I suggested is the better choice.

Reply:
That's sadly not possible, if it's `errorConfig` it is unexported and I can not access it in `span.go`. Any other idea?

Reply:
Sorry, I wasn't clear enough. What I meant was to just move these options in here:
```go
type FinishConfig struct {
	// FinishTime represents the time that should be set as finishing time for the
	// span. Implementations should use the current time when FinishTime.IsZero().
	FinishTime time.Time

	// Error holds an optional error that should be set on the span before
	// finishing.
	Error error

	// NoDebugStack will prevent any set errors from generating an attached stack trace tag.
	NoDebugStack bool

	// StackFrames specifies the number of stack frames to be attached in spans that finish with errors.
	StackFrames uint

	// StackSkip specifies the offset at which to start reporting stack frames from the stack.
	StackSkip uint
}
```
And then, adding a private `struct` in the `tracer` package:
```go
type errorConfig struct {
    noDebugStack bool
    stackFrames  uint
    stackSkip    uint
}
```
And simply using that as an argument to `setTagError`:
```go
func setTagError(v interface{}, cfg *errorConfig) {
```
I realise that this creates a bit of duplication, but I prefer that to making the public API confusing (`ErrorConfig` not used anywhere) just for the sake of simplifying internal methods.

If you feel otherwise or I might've missed something in my logic, please feel free to share.

Reply:
You would then call `setTagError` like this:
```go
s.setTagError(cfg.Error, &errorConfig{
    noDebugStack: cfg.NoDebugStack,
    stackFrames: cfg.StackFrames,
    stackSkip: cfg.StackSkip,
})
```